src/annotator/test/frame-observer-test.js

```diff
@@ -4,140 +4,191 @@ import {
+      iframe.removeAttribute('enable-annotation');
+      await waitForFrameObserver();
 
-    assert.calledOnce(onFrameRemoved);
-    assert.calledWith(onFrameRemoved, frame);
-  });
+      assert.calledOnce(onFrameRemoved);
+      assert.calledWith(onFrameRemoved, iframe);
+    });
 
-  it('changing the `src` attribute triggers onFrameRemoved', async () => {
-    const frame = createAnnotatableIFrame();
-    await waitForFrameObserver();
+    it('changing the `src` attribute triggers onFrameRemoved', async () => {
+      const iframe = createAnnotatableIFrame();
+      await waitForFrameObserver();
 
-    assert.calledOnce(onFrameAdded);
-    assert.calledWith(onFrameAdded, frame);
+      assert.calledOnce(onFrameAdded);
+      assert.calledWith(onFrameAdded, iframe);
 
-    frame.setAttribute('src', document.location);
-    await waitForIFrameUnload(frame);
+      iframe.setAttribute('src', document.location);
+      await waitForIFrameUnload(iframe);
 
-    assert.calledOnce(onFrameRemoved);
-    assert.calledWith(onFrameRemoved, frame);
-  });
+      assert.calledOnce(onFrameRemoved);
+      assert.calledWith(onFrameRemoved, iframe);
+    });
+
+    it(`doesn't call onFrameAdded if FrameObserver is disconnected`, async () => {
+      frameObserver.disconnect();
+      const iframe = createAnnotatableIFrame();
+
+      frameObserver._discoverFrames(); // Emulate a race condition
+      await onDocumentReady(iframe);
 
-  it(`doesn't call onFrameAdded if FrameObserver is disconnected`, async () => {
-    frameObserver.disconnect();
-    const frame = createAnnotatableIFrame();
+      assert.notCalled(onFrameAdded);
+    });
 
-    frameObserver._discoverFrames(); // Emulate a race condition
-    await onDocumentReady(frame);
+    it("doesn't trigger onFrameAdded when annotatable iframe is from a different domain", async () => {
+      const iframe = createAnnotatableIFrame();
+      iframe.setAttribute('src', 'http://cross-origin.dummy');
 
-    assert.notCalled(onFrameAdded);
+      await onDocumentReady(iframe);
+      await waitForFrameObserver();
+
+      assert.notCalled(onFrameAdded);
+      assert.calledOnce(console.warn);
+    });
   });
 
-  // This test doesn't work. Surprisingly, `isAccessible` returns `true` even
-  // thought the iframe is from a different domain. My suspicion is that the
-  // iframe is accessed at a time where the loading has not yet started and it
-  // is therefore accessible.
-  // it("doesn't triggers onFrameAdded when annotatable iframe is from a different domain", async () => {
-  //   const frame = createAnnotatableIFrame('src', 'https://example.com');
-  //   frame.setAttribute('enable-annotation', '');
-  //   await waitForFrameObserver();
-
-  //   assert.notCalled(onFrameAdded);
-  // });
+  describe('onDocumentReady', () => {
+    let fakeIFrame;
+    let fakeIFrameDocument;
+
+    beforeEach(() => {
+      fakeIFrameDocument = {
+        addEventListener: sinon.stub(),
```

Comment:
In recent versions of a Chrome a more convenient way to make a fake browser object that handles events is to create a new `EventTarget` instance or subclass `EventTarget`.

Reply:
👍🏼 . Incorporated this feedback in the last commit.